execution/evm/stack.go

```diff
@@ -185,3 +189,29 @@ func Is64BitOverflow(word Word256) bool {
 	}
 	return false
 }
+
+func (st *Stack) tryGrow() bool {
+	stackCap := cap(st.data)
+	hasMaxCap := st.maxCapacity > 0
+	if hasMaxCap && st.maxCapacity == stackCap {
+		return false
+	}
+
+	newCap := stackCap * 2
+	if newCap == 0 {
+		// Initialized with no capacity but stack can grow
+		newCap = 2
+	}
+
+	if hasMaxCap && st.maxCapacity < newCap {
+		// Last attempt to grow
+		newCap = st.maxCapacity
+	}
+
+	// Grow
+	newData := make([]Word256, newCap)
+	copy(newData, st.data)
+	st.data = newData
+
+	return true
```

Comment:
This whole function basically looks fine to me, but it occurs to me are doing basically the same thing here, but in a slightly different way: https://github.com/hyperledger/burrow/blob/develop/execution/evm/memory.go#L111-L142 

Thanks to Go and its type system we can't really reuse the implementation, but I wonder if we might want to borrow a few things from it:

1. It relies on the Go runtime to grow the backing array - this ends up being exponentially basically like you have done but has some optimisation for small sizes. I can't say it would make a dramatic difference, but I'd be inclined to piggy-back on Go's growth strategy which is more heavily optimised (and we get get at by using `append`).

2. There are some overflow checks there that we don't have here.